frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchAction.java

```diff
@@ -59,16 +63,21 @@ public void actionPerformed(AnActionEvent anActionEvent) {
     log().debug("Performing");
 
     var project = getProject(anActionEvent);
-    var gitRepository = getSelectedGitRepository(anActionEvent);
-    var branchName = getNameOfBranchUnderAction(anActionEvent);
-
-    if (branchName.isDefined() && gitRepository.isDefined()) {
-      doPull(project, gitRepository.get(), branchName.get());
-      getGraphTable(anActionEvent).queueRepositoryUpdateAndModelRefresh();
+    var gitRepository = getSelectedGitRepository(anActionEvent).getOrNull();
+    var branchName = getNameOfBranchUnderAction(anActionEvent).getOrNull();
+
+    if (branchName != null && gitRepository != null) {
+      var name = Option.of(gitRepository.getCurrentBranch()).map(b -> b.getName()).getOrNull();
+      if (branchName.equals(name)) {
+        doPull(project, gitRepository, branchName);
```

Comment:
Hmm why specifically not graph table update in case of pulling the current branch? :thinking: maybe drop a comment

Reply:
Actually we can just depend on the topic `GitRepository.GIT_REPO_CHANGE` in both cases